tests/standalone_tests/test_celery_app.py

```diff
@@ -0,0 +1,389 @@
+from tests.standalone_tests.conftest import ensure_localnodetmp_node_service_is_running
+from tests.standalone_tests.conftest import kill_service
+from tests.standalone_tests.conftest import remove_localnodetmp_rabbitmq
+from tests.standalone_tests.nodes_communication_helper import get_celery_task_signature
+from tests.standalone_tests.test_udfs import create_table_with_one_column_and_ten_rows
+
+request_id = "TestingCeleryApp"
+GET_NODE_INFO_TASK_TIMEOUT = 2
+
+
+def send_get_node_info_task_and_assert_response(
+    cel_app_wrapper, controller_testing_logger
+):
+    async_res = cel_app_wrapper.queue_task(
+        task_signature=get_celery_task_signature("get_node_info"),
+        logger=controller_testing_logger,
+        request_id=request_id,
+    )
+    res = cel_app_wrapper.get_result(
+        async_result=async_res,
+        timeout=GET_NODE_INFO_TASK_TIMEOUT,
+        logger=controller_testing_logger,
+    )
+    assert NodeInfo.parse_raw(res)
+
+
+def reset_the_celery_app_by_submitting_a_task_that_will_fail(
+    cel_app_wrapper, controller_testing_logger
+):
+    # Due to a bug on the rpc, the rabbitmq cannot recover the result, only the next scheduled task will be able to.
+    async_res = cel_app_wrapper.queue_task(
+        task_signature=get_celery_task_signature("get_node_info"),
+        logger=controller_testing_logger,
+        request_id=request_id,
+    )
+    with pytest.raises(CeleryConnectionError):
+        cel_app_wrapper.get_result(
+            async_result=async_res,
+            timeout=GET_NODE_INFO_TASK_TIMEOUT,
+            logger=controller_testing_logger,
+        )
+
+
+def queue_slow_udf(cel_app, logger):
+    run_udf_task = get_celery_task_signature("run_udf")
+    input_table_name, input_table_name_sum = create_table_with_one_column_and_ten_rows(
+        cel_app
+    )
+    kw_args_str = UDFKeyArguments(
+        args={"table": NodeTableDTO(value=input_table_name)}
+    ).json()
+
+    return cel_app.queue_task(
+        task_signature=run_udf_task,
+        logger=logger,
+        request_id=request_id,
+        command_id="1",
+        context_id=request_id,
+        func_name=make_unique_func_name(very_slow_udf),
+        positional_args_json=UDFPosArguments(args=[]).json(),
+        keyword_args_json=kw_args_str,
+    )
+
+
+@pytest.mark.slow
+@pytest.mark.integration
+def test_celery_app_queue_and_get_task(
+    globalnode_node_service,
+    reset_celery_app_factory,
+    controller_testing_logger,
+):
+    cel_app_wrapper = CeleryAppFactory().get_celery_app(RABBITMQ_GLOBALNODE_ADDR)
+    initial_cel_app = cel_app_wrapper._celery_app
+
+    send_get_node_info_task_and_assert_response(
+        cel_app_wrapper, controller_testing_logger
+    )
+
+    assert (
+        initial_cel_app
+        == CeleryAppFactory().get_celery_app(RABBITMQ_GLOBALNODE_ADDR)._celery_app
+    ), "Celery app is different after the task was queued and fetched."
```

Comment:
Inside the `send_get_node_info_task_and_assert_response` the response of the task is also asserted.

Asserting the cel_app as well is to make sure there were no unwanting changes by calling the task.

Reply:
I would either break this check into more than one tests, or at least(since ok, you essentially have to follow the same steps but check for different things) comment why this check is placed there (and all the other similar tests), because it would not make much sense for anyone else looking around these tests

Reply:
All the other test cases have fail messages that are self explanatory of why the check is there.

I will change it on this test as well to be more clear.

Reply:
I disagree with the logic that only one thing should be tested in each test.

These are not cheap unit tests, these are integration tests that normally take 1 minute to load.

These tests test application behavior on major scenarios that we have. It wouldn't offer anything to break it into code specific tests...

My understanding of how tests should be is pretty much summarized here: 
https://stackoverflow.com/questions/155436/unit-test-naming-best-practices
https://enterprisecraftsmanship.com/posts/you-naming-tests-wrong/

Reply:
I have not thoroughly read your resources but I did not find someone stating anything about multiple or single assertions in tests. Nevertheless, having single assertion per tests 1.makes it easier to see what causes the fail on a test and 2.makes it easier for the reader to understand what is being tested.

Reply:
I think what is being tested should be visible in the test name and should be an application behavior, not test_that_x_plus_y_equals_z. 

When something fails, the message shown should be the explanation of why it failed and that's why all these asserts have fail messages explaining what happened.